build-tools/garbage-files-generator.py

```diff
@@ -0,0 +1,113 @@
+#!/usr/bin/env python
+
+import argparse
+import logging
+import math
+import os
+import pathlib
+import re
+
+from rich.progress import Progress
+
+
+logging.basicConfig(
+    level=logging.INFO,
+    format="%(asctime)s - %(levelname)-8s - %(filename)s:%(lineno)d |> %(message)s",
+)
+
+
+OUTPUT_FOLDER = pathlib.Path("./data")
+
+
+def main():
+    args = _parse_args()
+    generate_data(
+        args.total_size,
+        args.files_count,
+        args.branching_factor,
+    )
+
+
+def generate_data(
+    total_size: int,
+    files_count: int,
+    branching_factor: int,
+):
+    file_size_bytes = math.ceil(total_size / files_count)
+    tree_depth = math.floor(math.log(files_count, branching_factor))
+    name_length = len(str(branching_factor))
+    logging.info(
+        f"Generating {files_count} files {file_size_bytes} bytes each, "
+        f"which in total: {total_size} bytes."
+    )
+
+    buffer_size = min(file_size_bytes, 16 * 2 ** 20)  # 16MB at max
+    garbage = os.urandom(buffer_size)
+    write_iterations = file_size_bytes // buffer_size
+    tail_size = file_size_bytes % buffer_size
+
+    OUTPUT_FOLDER.mkdir(parents=True, exist_ok=True)
+
+    created_files = 0
+    folders_counter = 0
+    with Progress() as progress:
+        data_gen_task = progress.add_task(
+            "[green]Generating data...", total=files_count
+        )
+        file_gen_task = progress.add_task(
+            f"[cyan]Generating file...", total=write_iterations
+        )
+        while created_files < files_count:
+            files_count_z = str(folders_counter).zfill(name_length * tree_depth)
+
+            split_path = []
+            for level in range(tree_depth):
+                split_path.append(
+                    files_count_z[name_length * level : name_length * (level + 1)]
+                )
+
+            folder_path = OUTPUT_FOLDER.joinpath(*split_path)
+
+            folder_path.mkdir(parents=True, exist_ok=True)
+            folders_counter += 1
+
+            for i in range(branching_factor):
+                if created_files < files_count:
+                    file_name = str(i).zfill(name_length)
+                    full_file_name = folder_path / file_name
+                    with full_file_name.open("wb") as file:
+                        for iteration in range(write_iterations):
+                            file.write(garbage)
+                            progress.update(
+                                file_gen_task,
+                                completed=iteration,
+                                description=f"[cyan]Writing file {full_file_name}...",
+                            )
+                        if tail_size != 0:
+                            file.write(garbage[:tail_size])
+                    created_files += 1
+                    progress.advance(data_gen_task)
+                else:
+                    break
+    logging.info("Data generation completed.")
```

Comment:
Very minor: if you started to use rich -- please create a `rich.console.Console` and use `console.log()` instead of `logging.info()`.  Mixing two styles can make a weird result eventually.

Reply:
Good idea, I replaced logging with rich.Console in 8c089ba.